File: Data Structures & Algorithms/subarray-sum-equals-k/submission-3.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int s = 0;
        int res = 0;
        HashMap<Integer, Integer> map = new HashMap<>();
        map.put(0, 1);
        for (int num : nums) {
            s += num;
            int dif = s - k;
            res += map.getOrDefault(dif, 0);
            map.put(s, map.getOrDefault(s, 0) + 1);
        }
        return res;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        std::unordered_map<int, int> count;
        vector<vector<int>> freq(nums.size() + 1);
        for (int num : nums) {
            count[num]++;
        }
        for (const auto& [x, y]: count) {
            freq[y].push_back(x);
        }
        vector<int> res;
        for (int i = freq.size() - 1; i > 0; --i) {
            for (int n : freq[i]) {
                res.push_back(n);
                if (res.size() == k) {
                    return res;
                }
            }
        }
        return res;
    }
};
